pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -255,11 +259,28 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	if node.Spec.ProviderID == "" {
 		return fmt.Errorf("node %s doesn't have providerID", nodeName)
 	}
-
-	cidrStrings, err := ca.cloud.AliasRangesByProviderID(node.Spec.ProviderID)
+	// fetch gce instance
+	instance, err := ca.cloud.InstanceByProviderID(node.Spec.ProviderID)
 	if err != nil {
 		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
-		return fmt.Errorf("failed to get cidr(s) from provider: %v", err)
+		return fmt.Errorf("failed to get instance from provider: %v", err)
+	}
+
+	cidrStrings := make([]string, 0)
+	var northInterfaces networkv1.NorthInterfacesAnnotation
+	var additionalNodeNetworks networkv1.MultiNetworkAnnotation
+
+	// nodes in clusters WITHOUT multi-networking are expected to have only 1 network-interface with 1 alias IP range.
```

Comment:
naive question, is not possible for a multi-network cluster to have nodes matching this condition?

Reply:
what if `instance.NetworkInterfaces` or `instance.NetworkInterfaces[0].AliasIpRanges` == 0 ? is that handled by `PerformMultiNetworkCIDRAllocation`?

Reply:
1. Since nodes in every cluster (multi-networking enabled/disabled) will have 1 NIC for sure (connected to the default network), for us, handling multi-networking comes into picture only when there are multiple network interfaces. So if multi-networking enabled clusters do have only 1 interface, it's safe to assume it's for default network. cc: @sdmodi 
2. Yes, `PerformMultiNetworkCIDRAllocation` just exits gracefully returning empty CIDR lists.

Reply:
2. I found being explicit about it more solid, and validate it here instead of relying on `PerformMultiNetworkCIDRAllocation`, we remove this dependency and the contract to keep maintaining this `gracefully returning empty CIDR lists.`  ... it also eliminates the risk that method gets refactored and we forget about this

Reply:
Sure, added a check.